packages/plugin-react/src/index.ts

```diff
@@ -114,54 +49,27 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
   const importReactRE = /\bimport\s+(?:\*\s+as\s+)?React\b/
 
   const viteBabel: Plugin = {
-    name: 'vite:react-babel',
-    enforce: 'pre',
-    config() {
-      if (opts.jsxRuntime === 'classic') {
-        return {
-          esbuild: {
-            jsx: 'transform',
-          },
-        }
-      } else {
-        return {
-          esbuild: {
-            jsx: 'automatic',
-            jsxImportSource: opts.jsxImportSource,
+    name: 'vite:react',
+    config(config, env) {
+      const runtime = opts.jsxRuntime ?? 'automatic'
+      return {
+        oxc: {
+          jsx: {
+            runtime,
+            importSource: runtime === 'automatic' ? jsxImportSource : undefined,
+            refresh: env.command === 'serve',
+            development: env.command === 'serve',
```

Comment:
Here enabled the rolldown-vite internal oxc transformer could be work. But here has some edge case.
- ssr need to disable refresh transformer
- the `js` `mdx` extension should be enabled refresh transformer

Here couldn't fine-grained control it. So i will using rolldown-vite exported `transformWithOxc` to instead of changed options.